tests/api2/test_261_iscsi_cmd.py

```diff
@@ -2491,6 +2494,48 @@ def test_28_portal_access(request):
                             assert MB_100 == _read_capacity16(s)
 
 
+def test_29_multiple_extents(request):
+    """
+    Verify that an iSCSI client can access multiple target LUNs
+    when multiple extents are configured.
+
+    Also validate that an extent serial number cannot be reused, and
+    that supplying an empty string serial number means one gets
+    generated.
+    """
+    iqn = f'{basename}:{target_name}'
+    with initiator_portal() as config:
+        portal_id = config['portal']['id']
+        with target(target_name, [{'portal': portal_id}]) as target_config:
+            target_id = target_config['id']
+            with dataset(pool_name, dataset_name):
+                with file_extent(pool_name, dataset_name, "target.extent1", filesize=MB_100, extent_name="extent1") as extent1_config:
+                    with file_extent(pool_name, dataset_name, "target.extent2", filesize=MB_256, extent_name="extent2") as extent2_config:
+                        with target_extent_associate(target_id, extent1_config['id'], 0):
+                            with target_extent_associate(target_id, extent2_config['id'], 1):
+                                with iscsi_scsi_connection(ip, iqn, 0) as s:
+                                    TUR(s)
+                                    s.blocksize = 512
+                                    assert MB_100 == _read_capacity16(s)
+                                with iscsi_scsi_connection(ip, iqn, 1) as s:
+                                    TUR(s)
+                                    s.blocksize = 512
+                                    assert MB_256 == _read_capacity16(s)
+
+                                # Now try to create another extent using the same serial number
+                                # We expect this to fail.
+                                with pytest.raises(AssertionError) as ve:
+                                    with file_extent(pool_name, dataset_name, "target.extent3", filesize=MB_512,
+                                                     extent_name="extent3", serial=extent1_config['serial']):
+                                        assert False, "Should not have been able to duplicate extent serial."
```

Comment:
You don't need to do this, `pytest.raise` will fail the test if the exception was not raised by the end of the block

Reply:
Agreed.  Thanks.